client/src/pages/ResourcePage.js

```diff
@@ -27,51 +37,33 @@ function createResourceElement(resource) {
   );
 }
 
-export default class ResourcePage extends React.Component {
-  /* 
-    Because we are using Axios to asynchronously fetch data from the API we have 
-    to handle the response and rendering of that data uniquely. 
-    We render once with initial settings and empty resources to set the UI.
-    Then we update state which will cause a second render once we have data.
-  */
-  constructor(props) {
-    super(props);
-    this.state = {
-      resources: [],
-      isError: false, // error fetching
-      isFetching: true // TODO: Instead of text we can have some loading animations
-    };
-  }
+// Function based component to use Redux easier.
+export default function ResourcePage() {
```

Comment:
the biggest difference here is i turned the component into a function comp instead of a class-based comp. this works better with redux.

instead of a constructor that sets this.state = {}, etc, now we want to use "useState" for local components. all of these fields should also be declared with a related setter, as you can see with the example "flags".

similar to useState we have:
- useSelector to get variables from the redux store to use in the comp, watch for changes, and re-render the comp on changes
- useDispatch to be able to dispatch actions and fetchers 
- useEffect: not too clear on what this is but seems to work like componentDidMount to run something after first render completes

Reply:
this pattern of redux functions can be used anywhere we want to use the redux store, such as sub components of the resources page. we can technically still keep using the old this.state constructor stuff, but it might be better to change everything everywhere to use "useState", even if it doesnt use the store.